presto-prometheus/src/main/java/com/facebook/presto/plugin/prometheus/PrometheusClient.java

```diff
@@ -148,17 +169,61 @@ public byte[] fetchUri(URI uri)
     {
         Request.Builder requestBuilder = new Request.Builder().url(uri.toString());
         getBearerAuthInfoFromFile().map(bearerToken -> requestBuilder.header("Authorization", "Bearer " + bearerToken));
-
         Response response;
         try {
-            response = httpClient.newCall(requestBuilder.build()).execute();
-            if (response.isSuccessful() && response.body() != null) {
-                return response.body().bytes();
+            if (config.isTlsEnabled()) {
+                OkHttpClient httpClient;
+                HostnameVerifier hostnameVerifier = new HostnameVerifier() {
+                    @Override
+                    public boolean verify(String hostname, SSLSession session)
+                    {
+                        // Always return true to accept any hostname
+                        return true;
+                    }
+                };
+                if (!config.getVerifyHostName()) {
+                    httpClient = new OkHttpClient.Builder()
+                            .hostnameVerifier(hostnameVerifier)
+                            .sslSocketFactory(getSSLContext().getSocketFactory(), (X509TrustManager) getTrustManagerFactory().getTrustManagers()[0])
+                            .build();
+                }
+                else {
+                    httpClient = new OkHttpClient.Builder()
+                            .sslSocketFactory(getSSLContext().getSocketFactory(), (X509TrustManager) getTrustManagerFactory().getTrustManagers()[0])
+                            .build();
+                }
+                response = httpClient.newCall(requestBuilder.build()).execute();
+                if (response.isSuccessful() && response.body() != null) {
+                    return response.body().bytes();
+                }
+            }
+            else {
+                response = httpClient.newCall(requestBuilder.build()).execute();
+                if (response.isSuccessful() && response.body() != null) {
+                    return response.body().bytes();
+                }
             }
         }
+        catch (SSLHandshakeException e) {
+            throw new PrestoException(SSL_HANDSHAKE_ERROR, "An SSL handshake error occurred while establishing a secure connection. Try the following measures to resolve the error:\n\n" + "- Upload a valid SSL certificate for authentication\n- Verify the expiration status of the uploaded certificate.\n- If you are connecting with SSL, enable SSL on both ends of the connection.\n");
+        }
+        catch (SSLPeerUnverifiedException e) {
+            throw new PrestoException(PEER_VERIFY_ERROR, "Peer verification failed.Below measures may resolve the issue \n" +
```

Comment:
missing space after period
Below --> These
may --> might (I think, never sure about that one)